### PR TITLE
Initial Symbol table import from github.com/mono/gtk-sharp/generator

### PR DESCRIPTION
**Pull request by decriptor**
I just started bring in the old symbol table. It doesn't compile yet, but just wanted to share the initial work and get feedback.

### CODE REVIEW

**Comment by decriptor** [comment]
This depends on PR#13

File: src/Gir/Generation/IAccessor.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿
+namespace Gir
+{
+	public interface IAccessor
+	{
+		void WriteAccessors(System.IO.StreamWriter sw, string indentation, string field_name);
```
Comment (Therzok) [inline src/Gir/Generation/IAccessor.cs]: This should take an IndentWriter, and no indentation string.

File: src/Gir/SymbolTable.cs [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+		{
+			return this[c_type] is Callback;
+		}
+
+		//public bool IsManuallyWrapped(string c_type)
+		//{
+		//	return this[c_type] is ManualGen;
+		//}
+
+		public string MangleName(string name)
```
Comment (Therzok) [inline src/Gir/SymbolTable.cs]: I'd rather not do this anymore, and actually `@` prefix keywords

File: src/Gir/SymbolTable.cs [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+			//AddType(new ConstStringGen("const-gchar"));
+			//AddType(new ConstStringGen("const-xmlChar"));
+			//AddType(new ConstStringGen("const-char"));
+			//AddType(new ConstFilenameGen("const-gfilename"));
+			//AddType(new StringMarshalGen("gfilename", "string", "IntPtr", "GLib.Marshaller.StringToFilenamePtr({0})", "GLib.Marshaller.FilenamePtrToStringGFree({0})"));
+			//AddType(new StringMarshalGen("gchar", "string", "IntPtr", "GLib.Marshaller.StringToPtrGStrdup({0})", "GLib.Marshaller.PtrToStringGFree({0})"));
+			//AddType(new StringMarshalGen("char", "string", "IntPtr", "GLib.Marshaller.StringToPtrGStrdup({0})", "GLib.Marshaller.PtrToStringGFree({0})"));
+			AddType(new Primitive("GStrv", "string[]", "null"));
+
+			// manually wrapped types requiring more complex marshaling
```
Comment (stsundermann) [inline src/Gir/SymbolTable.cs]: I don't think we want any of these from here on. Some are not bound correctly this way and we can generate these from GLib.gir.

**Comment by Therzok** [comment]
I'm going to take a bit of a different approach to importing symbol table.

File: src/Gir/SymbolTable.cs (hunk 3 above)
Comment (decriptor) [reply src/Gir/SymbolTable.cs]: IF we can pick these up from the gir files I think that is a much better idea.

Comment (stsundermann) [reply src/Gir/SymbolTable.cs]: All of them are in the gir files. We can even generate the whole GLib if the generator is complete and works. We can then add .NET integration with these classes on top of the generated classes.

**Comment by decriptor** [comment]
I'm going to close this in favor of https://github.com/mono/gir-sharp/pull/18

**Pull request closed without merging**